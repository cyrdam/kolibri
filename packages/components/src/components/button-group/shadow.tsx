import type { ButtonGroupProps } from '../../schema';
import type { JSX } from '@stencil/core';
import { Component, h, Host } from '@stencil/core';
import { KolButtonGroupWcTag } from '../../core/component-names';

@Component({
	tag: 'kol-button-group',
	styleUrls: {
		default: './style.scss',
	},
	shadow: true,
})
export class KolButtonGroup implements ButtonGroupProps {
	public render(): JSX.Element {
		return (
			<Host class="kol-button-group">
				<KolButtonGroupWcTag>
					<slot />
				</KolButtonGroupWcTag>
			</Host>
		);
	}
}
