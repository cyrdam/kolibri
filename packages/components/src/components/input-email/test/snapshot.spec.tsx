import { executeTests } from 'stencil-awesome-test';

import { h } from '@stencil/core';
import { newSpecPage } from '@stencil/core/testing';

import { getInputEmailHtml } from './html.mock';

import type { SpecPage } from '@stencil/core/testing';
import type { InputEmailProps } from '../../../schema';
import { KolInputEmail } from '../shadow';

executeTests<InputEmailProps>(
	'InputEmail',
	async (props): Promise<SpecPage> => {
		const page = await newSpecPage({
			components: [KolInputEmail],
			template: () => <kol-input-email {...props} />,
		});
		return page;
	},
	{
		_label: ['Label'],
		_hideLabel: [true, false],
		_disabled: [true, false],
		_alert: [true, false],
		_icons: [[{ left: 'codicon codicon-home' }]],
		_readOnly: [true, false],
		_msg: [{ _type: 'error', _description: 'Error message' }],
		_required: [true, false],
		_touched: [true, false],
		_smartButton: [
			{
				_icons: ['codicon codicon-eye'],
				_hideLabel: true,
				_label: 'einblenden',
			},
		],
	},
	getInputEmailHtml,
	{
		execMode: 'default', // ready
		needTimers: true,
	},
);
