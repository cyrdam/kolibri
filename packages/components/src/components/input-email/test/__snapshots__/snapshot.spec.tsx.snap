// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`kol-input-email should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value="email@example.com" _accessKey="V" 1`] = `
<kol-input-email>
  <template shadowrootmode="open">
    <div class="email has-value kol-form-field kol-input-email" role="presentation">
      <label class="kol-form-field__label" htmlfor="id-nonce" id="id-nonce-label">
        <span class="kol-form-field__label-text">
          Label
          <span aria-hidden="true" class="access-key-hint">
            V
          </span>
        </span>
      </label>
      <div class="kol-form-field__input">
        <div class="kol-input-container">
          <div class="kol-input-container__container">
            <input accesskey="V" autocapitalize="off" autocomplete="off" autocorrect="off" class="kol-input" id="id-nonce" name="field" placeholder="Hier steht ein Platzhaltertext" type="email" value="email@example.com">
          </div>
        </div>
      </div>
    </div>
  </template>
</kol-input-email>
`;

exports[`kol-input-email should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value="email@example.com" _alert=true 1`] = `
<kol-input-email _alert="">
  <template shadowrootmode="open">
    <div class="email has-value kol-form-field kol-input-email" role="presentation">
      <label class="kol-form-field__label" htmlfor="id-nonce" id="id-nonce-label">
        <span class="kol-form-field__label-text">
          <span>
            Label
          </span>
        </span>
      </label>
      <div class="kol-form-field__input">
        <div class="kol-input-container">
          <div class="kol-input-container__container">
            <input autocapitalize="off" autocomplete="off" autocorrect="off" class="kol-input" id="id-nonce" name="field" placeholder="Hier steht ein Platzhaltertext" type="email" value="email@example.com">
          </div>
        </div>
      </div>
    </div>
  </template>
</kol-input-email>
`;

exports[`kol-input-email should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value="email@example.com" _disabled=true 1`] = `
<kol-input-email>
  <template shadowrootmode="open">
    <div class="email has-value kol-form-field kol-form-field--disabled kol-input-email" role="presentation">
      <label class="kol-form-field__label" htmlfor="id-nonce" id="id-nonce-label">
        <span class="kol-form-field__label-text">
          <span>
            Label
          </span>
        </span>
      </label>
      <div class="kol-form-field__input">
        <div class="kol-input-container kol-input-container--disabled">
          <div class="kol-input-container__container">
            <input autocapitalize="off" autocomplete="off" autocorrect="off" class="kol-input kol-input--disabled" disabled="" id="id-nonce" name="field" placeholder="Hier steht ein Platzhaltertext" type="email" value="email@example.com">
          </div>
        </div>
      </div>
    </div>
  </template>
</kol-input-email>
`;

exports[`kol-input-email should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value="email@example.com" _hint="Hint" 1`] = `
<kol-input-email>
  <template shadowrootmode="open">
    <div class="email has-value kol-form-field kol-input-email" role="presentation">
      <label class="kol-form-field__label" htmlfor="id-nonce" id="id-nonce-label">
        <span class="kol-form-field__label-text">
          <span>
            Label
          </span>
        </span>
      </label>
      <div class="kol-form-field__input">
        <div class="kol-input-container">
          <div class="kol-input-container__container">
            <input aria-describedby="id-nonce-hint" autocapitalize="off" autocomplete="off" autocorrect="off" class="kol-input" id="id-nonce" name="field" placeholder="Hier steht ein Platzhaltertext" type="email" value="email@example.com">
          </div>
        </div>
      </div>
      <span class="kol-form-field__hint" id="id-nonce-hint">
        Hint
      </span>
    </div>
  </template>
</kol-input-email>
`;

exports[`kol-input-email should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value="email@example.com" _icons={"left":{"icon":"codicon codicon-arrow-left"},"right":{"icon":"codicon codicon-arrow-right"}} 1`] = `
<kol-input-email>
  <template shadowrootmode="open">
    <div class="email has-value kol-form-field kol-input-email" role="presentation">
      <label class="kol-form-field__label" htmlfor="id-nonce" id="id-nonce-label">
        <span class="kol-form-field__label-text">
          <span>
            Label
          </span>
        </span>
      </label>
      <div class="kol-form-field__input">
        <div class="kol-input-container">
          <div class="kol-input-container__adornment kol-input-container__adornment--start">
            <kol-icon _icons="codicon codicon-codicon codicon-arrow-left" _label=""></kol-icon>
          </div>
          <div class="kol-input-container__container">
            <input autocapitalize="off" autocomplete="off" autocorrect="off" class="kol-input" id="id-nonce" name="field" placeholder="Hier steht ein Platzhaltertext" type="email" value="email@example.com">
          </div>
          <div class="kol-input-container__adornment kol-input-container__adornment--end">
            <kol-icon _icons="codicon codicon-codicon codicon-arrow-right" _label=""></kol-icon>
          </div>
        </div>
      </div>
    </div>
  </template>
</kol-input-email>
`;

exports[`kol-input-email should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value="email@example.com" _icons={"left":{"icon":"codicon codicon-arrow-left"}} 1`] = `
<kol-input-email>
  <template shadowrootmode="open">
    <div class="email has-value kol-form-field kol-input-email" role="presentation">
      <label class="kol-form-field__label" htmlfor="id-nonce" id="id-nonce-label">
        <span class="kol-form-field__label-text">
          <span>
            Label
          </span>
        </span>
      </label>
      <div class="kol-form-field__input">
        <div class="kol-input-container">
          <div class="kol-input-container__adornment kol-input-container__adornment--start">
            <kol-icon _icons="codicon codicon-codicon codicon-arrow-left" _label=""></kol-icon>
          </div>
          <div class="kol-input-container__container">
            <input autocapitalize="off" autocomplete="off" autocorrect="off" class="kol-input" id="id-nonce" name="field" placeholder="Hier steht ein Platzhaltertext" type="email" value="email@example.com">
          </div>
          <div class="kol-input-container__adornment kol-input-container__adornment--end"></div>
        </div>
      </div>
    </div>
  </template>
</kol-input-email>
`;

exports[`kol-input-email should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value="email@example.com" _icons={"right":{"icon":"codicon codicon-arrow-right"}} 1`] = `
<kol-input-email>
  <template shadowrootmode="open">
    <div class="email has-value kol-form-field kol-input-email" role="presentation">
      <label class="kol-form-field__label" htmlfor="id-nonce" id="id-nonce-label">
        <span class="kol-form-field__label-text">
          <span>
            Label
          </span>
        </span>
      </label>
      <div class="kol-form-field__input">
        <div class="kol-input-container">
          <div class="kol-input-container__adornment kol-input-container__adornment--start"></div>
          <div class="kol-input-container__container">
            <input autocapitalize="off" autocomplete="off" autocorrect="off" class="kol-input" id="id-nonce" name="field" placeholder="Hier steht ein Platzhaltertext" type="email" value="email@example.com">
          </div>
          <div class="kol-input-container__adornment kol-input-container__adornment--end">
            <kol-icon _icons="codicon codicon-codicon codicon-arrow-right" _label=""></kol-icon>
          </div>
        </div>
      </div>
    </div>
  </template>
</kol-input-email>
`;

exports[`kol-input-email should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value="email@example.com" _msg={"_type":"error","_description":"Es ist ein Fehler aufgetreten"} _hideError=true 1`] = `
<kol-input-email _hide-error="">
  <template shadowrootmode="open">
    <div class="email has-value kol-form-field kol-form-field--hidden-error kol-input-email" role="presentation">
      <label class="kol-form-field__label" htmlfor="id-nonce" id="id-nonce-label">
        <span class="kol-form-field__label-text">
          <span>
            Label
          </span>
        </span>
      </label>
      <div class="kol-form-field__input">
        <div class="kol-input-container">
          <div class="kol-input-container__container">
            <input autocapitalize="off" autocomplete="off" autocorrect="off" class="kol-input" id="id-nonce" name="field" placeholder="Hier steht ein Platzhaltertext" type="email" value="email@example.com">
          </div>
        </div>
      </div>
    </div>
  </template>
</kol-input-email>
`;

exports[`kol-input-email should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value="email@example.com" _msg={"_type":"error","_description":"Es ist ein Fehler aufgetreten"} 1`] = `
<kol-input-email>
  <template shadowrootmode="open">
    <div class="email has-value kol-form-field kol-input-email" role="presentation">
      <label class="kol-form-field__label" htmlfor="id-nonce" id="id-nonce-label">
        <span class="kol-form-field__label-text">
          <span>
            Label
          </span>
        </span>
      </label>
      <div class="kol-form-field__input">
        <div class="kol-input-container">
          <div class="kol-input-container__container">
            <input autocapitalize="off" autocomplete="off" autocorrect="off" class="kol-input" id="id-nonce" name="field" placeholder="Hier steht ein Platzhaltertext" type="email" value="email@example.com">
          </div>
        </div>
      </div>
    </div>
  </template>
</kol-input-email>
`;

exports[`kol-input-email should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value="email@example.com" _readOnly=true 1`] = `
<kol-input-email>
  <template shadowrootmode="open">
    <div class="email has-value kol-form-field kol-form-field--read-only kol-input-email" role="presentation">
      <label class="kol-form-field__label" htmlfor="id-nonce" id="id-nonce-label">
        <span class="kol-form-field__label-text">
          <span>
            Label
          </span>
        </span>
      </label>
      <div class="kol-form-field__input">
        <div class="kol-input-container">
          <div class="kol-input-container__container">
            <input autocapitalize="off" autocomplete="off" autocorrect="off" class="kol-input kol-input--readonly" id="id-nonce" name="field" placeholder="Hier steht ein Platzhaltertext" readonly="" type="email" value="email@example.com">
          </div>
        </div>
      </div>
    </div>
  </template>
</kol-input-email>
`;

exports[`kol-input-email should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value="email@example.com" _shortKey="S" 1`] = `
<kol-input-email>
  <template shadowrootmode="open">
    <div class="email has-value kol-form-field kol-input-email" role="presentation">
      <label class="kol-form-field__label" htmlfor="id-nonce" id="id-nonce-label">
        <span class="kol-form-field__label-text">
          Label
          <span aria-hidden="true" class="access-key-hint">
            S
          </span>
        </span>
      </label>
      <div class="kol-form-field__input">
        <div class="kol-input-container">
          <div class="kol-input-container__container">
            <input autocapitalize="off" autocomplete="off" autocorrect="off" class="kol-input" id="id-nonce" name="field" placeholder="Hier steht ein Platzhaltertext" type="email" value="email@example.com">
          </div>
        </div>
      </div>
    </div>
  </template>
</kol-input-email>
`;

exports[`kol-input-email should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value="email@example.com" _smartButton={"_icons":["codicon codicon-eye"],"_hideLabel":true,"_label":"einblenden"} 1`] = `
<kol-input-email>
  <template shadowrootmode="open">
    <div class="email has-value kol-form-field kol-input-email" role="presentation">
      <label class="kol-form-field__label" htmlfor="id-nonce" id="id-nonce-label">
        <span class="kol-form-field__label-text">
          <span>
            Label
          </span>
        </span>
      </label>
      <div class="kol-form-field__input">
        <div class="kol-input-container">
          <div class="kol-input-container__adornment kol-input-container__adornment--start"></div>
          <div class="kol-input-container__container">
            <input autocapitalize="off" autocomplete="off" autocorrect="off" class="kol-input" id="id-nonce" name="field" placeholder="Hier steht ein Platzhaltertext" type="email" value="email@example.com">
          </div>
          <div class="kol-input-container__adornment kol-input-container__adornment--end">
            <kol-button-wc _hidelabel="" _label="einblenden" tabindex="0"></kol-button-wc>
          </div>
        </div>
      </div>
    </div>
  </template>
</kol-input-email>
`;

exports[`kol-input-email should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value="email@example.com" _suggestions=["email1@example.com","email2@example.com","email3@example.com"] _accessKey="V" 1`] = `
<kol-input-email>
  <template shadowrootmode="open">
    <div class="email has-value kol-form-field kol-input-email" role="presentation">
      <label class="kol-form-field__label" htmlfor="id-nonce" id="id-nonce-label">
        <span class="kol-form-field__label-text">
          Label
          <span aria-hidden="true" class="access-key-hint">
            V
          </span>
        </span>
      </label>
      <div class="kol-form-field__input">
        <div class="kol-input-container">
          <div class="kol-input-container__container">
            <input accesskey="V" autocapitalize="off" autocomplete="off" autocorrect="off" class="kol-input" id="id-nonce" list="id-nonce-list" name="field" placeholder="Hier steht ein Platzhaltertext" type="email" value="email@example.com">
            <datalist id="id-nonce-list">
              <option value="email1@example.com"></option>
              <option value="email2@example.com"></option>
              <option value="email3@example.com"></option>
            </datalist>
          </div>
        </div>
      </div>
    </div>
  </template>
</kol-input-email>
`;

exports[`kol-input-email should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value="email@example.com" _suggestions=["email1@example.com","email2@example.com","email3@example.com"] _alert=true 1`] = `
<kol-input-email _alert="">
  <template shadowrootmode="open">
    <div class="email has-value kol-form-field kol-input-email" role="presentation">
      <label class="kol-form-field__label" htmlfor="id-nonce" id="id-nonce-label">
        <span class="kol-form-field__label-text">
          <span>
            Label
          </span>
        </span>
      </label>
      <div class="kol-form-field__input">
        <div class="kol-input-container">
          <div class="kol-input-container__container">
            <input autocapitalize="off" autocomplete="off" autocorrect="off" class="kol-input" id="id-nonce" list="id-nonce-list" name="field" placeholder="Hier steht ein Platzhaltertext" type="email" value="email@example.com">
            <datalist id="id-nonce-list">
              <option value="email1@example.com"></option>
              <option value="email2@example.com"></option>
              <option value="email3@example.com"></option>
            </datalist>
          </div>
        </div>
      </div>
    </div>
  </template>
</kol-input-email>
`;

exports[`kol-input-email should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value="email@example.com" _suggestions=["email1@example.com","email2@example.com","email3@example.com"] _disabled=true 1`] = `
<kol-input-email>
  <template shadowrootmode="open">
    <div class="email has-value kol-form-field kol-form-field--disabled kol-input-email" role="presentation">
      <label class="kol-form-field__label" htmlfor="id-nonce" id="id-nonce-label">
        <span class="kol-form-field__label-text">
          <span>
            Label
          </span>
        </span>
      </label>
      <div class="kol-form-field__input">
        <div class="kol-input-container kol-input-container--disabled">
          <div class="kol-input-container__container">
            <input autocapitalize="off" autocomplete="off" autocorrect="off" class="kol-input kol-input--disabled" disabled="" id="id-nonce" list="id-nonce-list" name="field" placeholder="Hier steht ein Platzhaltertext" type="email" value="email@example.com">
            <datalist id="id-nonce-list">
              <option value="email1@example.com"></option>
              <option value="email2@example.com"></option>
              <option value="email3@example.com"></option>
            </datalist>
          </div>
        </div>
      </div>
    </div>
  </template>
</kol-input-email>
`;

exports[`kol-input-email should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value="email@example.com" _suggestions=["email1@example.com","email2@example.com","email3@example.com"] _hint="Hint" 1`] = `
<kol-input-email>
  <template shadowrootmode="open">
    <div class="email has-value kol-form-field kol-input-email" role="presentation">
      <label class="kol-form-field__label" htmlfor="id-nonce" id="id-nonce-label">
        <span class="kol-form-field__label-text">
          <span>
            Label
          </span>
        </span>
      </label>
      <div class="kol-form-field__input">
        <div class="kol-input-container">
          <div class="kol-input-container__container">
            <input aria-describedby="id-nonce-hint" autocapitalize="off" autocomplete="off" autocorrect="off" class="kol-input" id="id-nonce" list="id-nonce-list" name="field" placeholder="Hier steht ein Platzhaltertext" type="email" value="email@example.com">
            <datalist id="id-nonce-list">
              <option value="email1@example.com"></option>
              <option value="email2@example.com"></option>
              <option value="email3@example.com"></option>
            </datalist>
          </div>
        </div>
      </div>
      <span class="kol-form-field__hint" id="id-nonce-hint">
        Hint
      </span>
    </div>
  </template>
</kol-input-email>
`;

exports[`kol-input-email should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value="email@example.com" _suggestions=["email1@example.com","email2@example.com","email3@example.com"] _icons={"left":{"icon":"codicon codicon-arrow-left"},"right":{"icon":"codicon codicon-arrow-right"}} 1`] = `
<kol-input-email>
  <template shadowrootmode="open">
    <div class="email has-value kol-form-field kol-input-email" role="presentation">
      <label class="kol-form-field__label" htmlfor="id-nonce" id="id-nonce-label">
        <span class="kol-form-field__label-text">
          <span>
            Label
          </span>
        </span>
      </label>
      <div class="kol-form-field__input">
        <div class="kol-input-container">
          <div class="kol-input-container__adornment kol-input-container__adornment--start">
            <kol-icon _icons="codicon codicon-codicon codicon-arrow-left" _label=""></kol-icon>
          </div>
          <div class="kol-input-container__container">
            <input autocapitalize="off" autocomplete="off" autocorrect="off" class="kol-input" id="id-nonce" list="id-nonce-list" name="field" placeholder="Hier steht ein Platzhaltertext" type="email" value="email@example.com">
            <datalist id="id-nonce-list">
              <option value="email1@example.com"></option>
              <option value="email2@example.com"></option>
              <option value="email3@example.com"></option>
            </datalist>
          </div>
          <div class="kol-input-container__adornment kol-input-container__adornment--end">
            <kol-icon _icons="codicon codicon-codicon codicon-arrow-right" _label=""></kol-icon>
          </div>
        </div>
      </div>
    </div>
  </template>
</kol-input-email>
`;

exports[`kol-input-email should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value="email@example.com" _suggestions=["email1@example.com","email2@example.com","email3@example.com"] _icons={"left":{"icon":"codicon codicon-arrow-left"}} 1`] = `
<kol-input-email>
  <template shadowrootmode="open">
    <div class="email has-value kol-form-field kol-input-email" role="presentation">
      <label class="kol-form-field__label" htmlfor="id-nonce" id="id-nonce-label">
        <span class="kol-form-field__label-text">
          <span>
            Label
          </span>
        </span>
      </label>
      <div class="kol-form-field__input">
        <div class="kol-input-container">
          <div class="kol-input-container__adornment kol-input-container__adornment--start">
            <kol-icon _icons="codicon codicon-codicon codicon-arrow-left" _label=""></kol-icon>
          </div>
          <div class="kol-input-container__container">
            <input autocapitalize="off" autocomplete="off" autocorrect="off" class="kol-input" id="id-nonce" list="id-nonce-list" name="field" placeholder="Hier steht ein Platzhaltertext" type="email" value="email@example.com">
            <datalist id="id-nonce-list">
              <option value="email1@example.com"></option>
              <option value="email2@example.com"></option>
              <option value="email3@example.com"></option>
            </datalist>
          </div>
          <div class="kol-input-container__adornment kol-input-container__adornment--end"></div>
        </div>
      </div>
    </div>
  </template>
</kol-input-email>
`;

exports[`kol-input-email should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value="email@example.com" _suggestions=["email1@example.com","email2@example.com","email3@example.com"] _icons={"right":{"icon":"codicon codicon-arrow-right"}} 1`] = `
<kol-input-email>
  <template shadowrootmode="open">
    <div class="email has-value kol-form-field kol-input-email" role="presentation">
      <label class="kol-form-field__label" htmlfor="id-nonce" id="id-nonce-label">
        <span class="kol-form-field__label-text">
          <span>
            Label
          </span>
        </span>
      </label>
      <div class="kol-form-field__input">
        <div class="kol-input-container">
          <div class="kol-input-container__adornment kol-input-container__adornment--start"></div>
          <div class="kol-input-container__container">
            <input autocapitalize="off" autocomplete="off" autocorrect="off" class="kol-input" id="id-nonce" list="id-nonce-list" name="field" placeholder="Hier steht ein Platzhaltertext" type="email" value="email@example.com">
            <datalist id="id-nonce-list">
              <option value="email1@example.com"></option>
              <option value="email2@example.com"></option>
              <option value="email3@example.com"></option>
            </datalist>
          </div>
          <div class="kol-input-container__adornment kol-input-container__adornment--end">
            <kol-icon _icons="codicon codicon-codicon codicon-arrow-right" _label=""></kol-icon>
          </div>
        </div>
      </div>
    </div>
  </template>
</kol-input-email>
`;

exports[`kol-input-email should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value="email@example.com" _suggestions=["email1@example.com","email2@example.com","email3@example.com"] _msg={"_type":"error","_description":"Es ist ein Fehler aufgetreten"} _hideError=true 1`] = `
<kol-input-email _hide-error="">
  <template shadowrootmode="open">
    <div class="email has-value kol-form-field kol-form-field--hidden-error kol-input-email" role="presentation">
      <label class="kol-form-field__label" htmlfor="id-nonce" id="id-nonce-label">
        <span class="kol-form-field__label-text">
          <span>
            Label
          </span>
        </span>
      </label>
      <div class="kol-form-field__input">
        <div class="kol-input-container">
          <div class="kol-input-container__container">
            <input autocapitalize="off" autocomplete="off" autocorrect="off" class="kol-input" id="id-nonce" list="id-nonce-list" name="field" placeholder="Hier steht ein Platzhaltertext" type="email" value="email@example.com">
            <datalist id="id-nonce-list">
              <option value="email1@example.com"></option>
              <option value="email2@example.com"></option>
              <option value="email3@example.com"></option>
            </datalist>
          </div>
        </div>
      </div>
    </div>
  </template>
</kol-input-email>
`;

exports[`kol-input-email should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value="email@example.com" _suggestions=["email1@example.com","email2@example.com","email3@example.com"] _msg={"_type":"error","_description":"Es ist ein Fehler aufgetreten"} 1`] = `
<kol-input-email>
  <template shadowrootmode="open">
    <div class="email has-value kol-form-field kol-input-email" role="presentation">
      <label class="kol-form-field__label" htmlfor="id-nonce" id="id-nonce-label">
        <span class="kol-form-field__label-text">
          <span>
            Label
          </span>
        </span>
      </label>
      <div class="kol-form-field__input">
        <div class="kol-input-container">
          <div class="kol-input-container__container">
            <input autocapitalize="off" autocomplete="off" autocorrect="off" class="kol-input" id="id-nonce" list="id-nonce-list" name="field" placeholder="Hier steht ein Platzhaltertext" type="email" value="email@example.com">
            <datalist id="id-nonce-list">
              <option value="email1@example.com"></option>
              <option value="email2@example.com"></option>
              <option value="email3@example.com"></option>
            </datalist>
          </div>
        </div>
      </div>
    </div>
  </template>
</kol-input-email>
`;

exports[`kol-input-email should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value="email@example.com" _suggestions=["email1@example.com","email2@example.com","email3@example.com"] _readOnly=true 1`] = `
<kol-input-email>
  <template shadowrootmode="open">
    <div class="email has-value kol-form-field kol-form-field--read-only kol-input-email" role="presentation">
      <label class="kol-form-field__label" htmlfor="id-nonce" id="id-nonce-label">
        <span class="kol-form-field__label-text">
          <span>
            Label
          </span>
        </span>
      </label>
      <div class="kol-form-field__input">
        <div class="kol-input-container">
          <div class="kol-input-container__container">
            <input autocapitalize="off" autocomplete="off" autocorrect="off" class="kol-input kol-input--readonly" id="id-nonce" list="id-nonce-list" name="field" placeholder="Hier steht ein Platzhaltertext" readonly="" type="email" value="email@example.com">
            <datalist id="id-nonce-list">
              <option value="email1@example.com"></option>
              <option value="email2@example.com"></option>
              <option value="email3@example.com"></option>
            </datalist>
          </div>
        </div>
      </div>
    </div>
  </template>
</kol-input-email>
`;

exports[`kol-input-email should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value="email@example.com" _suggestions=["email1@example.com","email2@example.com","email3@example.com"] _shortKey="S" 1`] = `
<kol-input-email>
  <template shadowrootmode="open">
    <div class="email has-value kol-form-field kol-input-email" role="presentation">
      <label class="kol-form-field__label" htmlfor="id-nonce" id="id-nonce-label">
        <span class="kol-form-field__label-text">
          Label
          <span aria-hidden="true" class="access-key-hint">
            S
          </span>
        </span>
      </label>
      <div class="kol-form-field__input">
        <div class="kol-input-container">
          <div class="kol-input-container__container">
            <input autocapitalize="off" autocomplete="off" autocorrect="off" class="kol-input" id="id-nonce" list="id-nonce-list" name="field" placeholder="Hier steht ein Platzhaltertext" type="email" value="email@example.com">
            <datalist id="id-nonce-list">
              <option value="email1@example.com"></option>
              <option value="email2@example.com"></option>
              <option value="email3@example.com"></option>
            </datalist>
          </div>
        </div>
      </div>
    </div>
  </template>
</kol-input-email>
`;

exports[`kol-input-email should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value="email@example.com" _suggestions=["email1@example.com","email2@example.com","email3@example.com"] _smartButton={"_icons":["codicon codicon-eye"],"_hideLabel":true,"_label":"einblenden"} 1`] = `
<kol-input-email>
  <template shadowrootmode="open">
    <div class="email has-value kol-form-field kol-input-email" role="presentation">
      <label class="kol-form-field__label" htmlfor="id-nonce" id="id-nonce-label">
        <span class="kol-form-field__label-text">
          <span>
            Label
          </span>
        </span>
      </label>
      <div class="kol-form-field__input">
        <div class="kol-input-container">
          <div class="kol-input-container__adornment kol-input-container__adornment--start"></div>
          <div class="kol-input-container__container">
            <input autocapitalize="off" autocomplete="off" autocorrect="off" class="kol-input" id="id-nonce" list="id-nonce-list" name="field" placeholder="Hier steht ein Platzhaltertext" type="email" value="email@example.com">
            <datalist id="id-nonce-list">
              <option value="email1@example.com"></option>
              <option value="email2@example.com"></option>
              <option value="email3@example.com"></option>
            </datalist>
          </div>
          <div class="kol-input-container__adornment kol-input-container__adornment--end">
            <kol-button-wc _hidelabel="" _label="einblenden" tabindex="0"></kol-button-wc>
          </div>
        </div>
      </div>
    </div>
  </template>
</kol-input-email>
`;

exports[`kol-input-email should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value="email@example.com" _suggestions=["email1@example.com","email2@example.com","email3@example.com"] _touched=true 1`] = `
<kol-input-email _touched="">
  <template shadowrootmode="open">
    <div class="email has-value kol-form-field kol-form-field--touched kol-input-email" role="presentation">
      <label class="kol-form-field__label" htmlfor="id-nonce" id="id-nonce-label">
        <span class="kol-form-field__label-text">
          <span>
            Label
          </span>
        </span>
      </label>
      <div class="kol-form-field__input">
        <div class="kol-input-container">
          <div class="kol-input-container__container">
            <input autocapitalize="off" autocomplete="off" autocorrect="off" class="kol-input kol-input--touched" id="id-nonce" list="id-nonce-list" name="field" placeholder="Hier steht ein Platzhaltertext" type="email" value="email@example.com">
            <datalist id="id-nonce-list">
              <option value="email1@example.com"></option>
              <option value="email2@example.com"></option>
              <option value="email3@example.com"></option>
            </datalist>
          </div>
        </div>
      </div>
    </div>
  </template>
</kol-input-email>
`;

exports[`kol-input-email should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value="email@example.com" _suggestions=["email1@example.com","email2@example.com","email3@example.com"] 1`] = `
<kol-input-email>
  <template shadowrootmode="open">
    <div class="email has-value kol-form-field kol-input-email" role="presentation">
      <label class="kol-form-field__label" htmlfor="id-nonce" id="id-nonce-label">
        <span class="kol-form-field__label-text">
          <span>
            Label
          </span>
        </span>
      </label>
      <div class="kol-form-field__input">
        <div class="kol-input-container">
          <div class="kol-input-container__container">
            <input autocapitalize="off" autocomplete="off" autocorrect="off" class="kol-input" id="id-nonce" list="id-nonce-list" name="field" placeholder="Hier steht ein Platzhaltertext" type="email" value="email@example.com">
            <datalist id="id-nonce-list">
              <option value="email1@example.com"></option>
              <option value="email2@example.com"></option>
              <option value="email3@example.com"></option>
            </datalist>
          </div>
        </div>
      </div>
    </div>
  </template>
</kol-input-email>
`;

exports[`kol-input-email should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value="email@example.com" _touched=true 1`] = `
<kol-input-email _touched="">
  <template shadowrootmode="open">
    <div class="email has-value kol-form-field kol-form-field--touched kol-input-email" role="presentation">
      <label class="kol-form-field__label" htmlfor="id-nonce" id="id-nonce-label">
        <span class="kol-form-field__label-text">
          <span>
            Label
          </span>
        </span>
      </label>
      <div class="kol-form-field__input">
        <div class="kol-input-container">
          <div class="kol-input-container__container">
            <input autocapitalize="off" autocomplete="off" autocorrect="off" class="kol-input kol-input--touched" id="id-nonce" name="field" placeholder="Hier steht ein Platzhaltertext" type="email" value="email@example.com">
          </div>
        </div>
      </div>
    </div>
  </template>
</kol-input-email>
`;

exports[`kol-input-email should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value="email@example.com" 1`] = `
<kol-input-email>
  <template shadowrootmode="open">
    <div class="email has-value kol-form-field kol-input-email" role="presentation">
      <label class="kol-form-field__label" htmlfor="id-nonce" id="id-nonce-label">
        <span class="kol-form-field__label-text">
          <span>
            Label
          </span>
        </span>
      </label>
      <div class="kol-form-field__input">
        <div class="kol-input-container">
          <div class="kol-input-container__container">
            <input autocapitalize="off" autocomplete="off" autocorrect="off" class="kol-input" id="id-nonce" name="field" placeholder="Hier steht ein Platzhaltertext" type="email" value="email@example.com">
          </div>
        </div>
      </div>
    </div>
  </template>
</kol-input-email>
`;
